File: src/main/java/nl/hanze/week2/assignment1/RemoteControl/Television.java
package nl.hanze.week2.assignment1.RemoteControl;

// a comand receiver
public class Television implements ElectronicDevice {

    private int volume = 0;
    
    public void on() {
        
        System.out.println("TV is on");
        
    }

    public void off() {
        
        System.out.println("TV is off");
        
    }

    public void volumeUp() {
        
        volume++;
        
        System.out.println("TV Volume is at: " + volume);
        
    }

    public void volumenDown() {
        
        volume--;
        
        System.out.println("TV Volume is at: " + volume);
        
    }
    
}

File: src/main/java/nl/hanze/week2/assignment1/devices/Television.kt
package nl.hanze.week2.assignment1.devices

import nl.hanze.week2.assignment1.root.ElectronicDevice

// a comand receiver
class Television : ElectronicDevice {

    private var volume = 0

    override fun on() = println("TV is on")
    override fun off() = println("TV is off")
    override fun volumeUp() = println("TV Volume is at: ${++volume}")
    override fun volumeDown() = println("""TV Volume is at: ${volume--}""")
}